adobe/aem-core-cif-components
# CIF-1043 - Apply & remove coupons from cart

## Description

* Added `CouponForm` component to display form to apply a coupon to the cart.
* Added `CouponItem` component to display any coupon that was applied to the cart and allows to remove it again.
* Added separate error handling for coupons.

## Verification Steps
* Add a product to the cart and try to apply coupon `coupon10`. The coupon should be applied and the prices updated accordingly.
* Remove the coupon. The coupon should be removed from the cart and the prices updated accordingly.
* It should not be possible to enter a coupon that is shorter than 3 letters.
* Entering an invalid coupon (e.g. `coupon20`) should result in an error. It should be possible to normally add a valid coupon when the error is shown.

## How Has This Been Tested?

* Added unit tests

## Screenshots (if appropriate):

## Types of changes




## Checklist:
